### ISSUE TITLE
Mobile version of TinyMCE should respect the noneditable plugin when using Silver theme

### ISSUE THREAD

**Issue by Xoduz** [tinymce/tinymce#4848]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Noneditable elements are freely editable when using a mobile version of TinyMCE that uses Silver theme instead of full screen editor.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tinymce.com](http://fiddle.tinymce.com/) or similar.**
Setup TinyMCE with the noneditable plugin and Silver theme for mobile instead of full screen editor, and specify that elements of a certain class should be noneditable. 

```
tinymce.init({
  selector: "textarea",
  plugins: "noneditable",
  noneditable_noneditable_class: "mceNonEditable",
  mobile: {
    theme: 'silver',
  },
});
```

Using desktop browser, add some elements through source code that contain the specified noneditable class. Test in desktop browser and observe that noneditable elements are not editable. Then, switch to a mobile device and observe how the noneditable elements ARE freely and fully editable (bug).

Now, remove the silver theme from the mobile setup, and test again using the fullscreen editor in your mobile device. Observe how the noneditable areas ARE NOT editable (correct).

**What is the expected behavior?**
Expected behaviour is that the mobile version of TinyMCE respect the noneditable setup, even when using the Silver theme instead of fullscreen editor.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
TinyMCE v5.0.1 (2019-02-21)
Chrome 72.0.3626.119 (Official Build) (64-bit)
Opera 58.0.3135.68
Opera for Android v49.2.2361.134358
Chrome for Android v71.0.3578.99
Android 9

**Comment by lnewson** [comment]
Thanks for the report! Looking into this it isn't specifically related to the silver theme. Instead the plugins for mobile configuration are currently separate to the desktop (as they were in v4) and default to the same mobile theme settings as v4 (autosave, autolink & lists) since those are the only supported plugins for the mobile.

As such, the noneditable plugin cannot isn't support on mobile. In saying that however I'm not sure if that's intentional, so I'll have to check with the team and will get back to you.

**Comment by lnewson** [comment]
I checked with the team and the silver theme currently isn't supported in mobile mode unfortunately, so it is behaving correctly as those three plugins are the only supported plugins for mobile mode.

In which case, I'll turn this into a feature request to be able to support silver and additional plugins for mobile.

**Comment by Xoduz** [comment]
Fair enough. One question though - how come the fullscreen editor actually does respect noneditable areas if such a setting has been setup for TinyMCE in general (not specific to mobile), if the plugin is not supported?

**Comment by spocke** [comment]
The noneditable and fullscreen mode seems to be working fine together on desktop. I created a fiddle where both are enabled in v5: http://fiddle.tinymce.com/PCgaab if this isn't working for you then open another ticket with a fiddle and steps for us to reproduce it.

**Comment by sebi-v** [comment]
I have an issue related to silver theme in mobile as well. So tinymce stripes out some of the plugins if the divice is phone (in my case removes link, image and custom bittons). Where in inline mode they work ok, since tinymce does not modify the plugins property on the settings, since inline is not supported on mobile. It would be nice to handle the silver mobile theme as inline mode, as in our case we want to disable the mobile theme.

**Comment by Xoduz** [comment]
@spocke Yeah that works for me too. My issue is that it does not work with silver theme on mobile - which it turns out is an unsupported feature.

**Comment by TheSpyder** [comment]
We've realised that the way we turned mobile on by default in v5 has changed the plugin defaults as well. Even though using desktop themes on mobile isn't something we test, we're going to log a task to link the plugin whitelist to the _theme_ instead of the _platform_ which should allow v4-like behaviour with using desktop themes on mobile if desired.

**Comment by TheSpyder** [comment]
The task hasn't been scheduled yet, sorry. There isn't any way to work around the code, the plugin list on mobile is filtered using this array:
https://github.com/tinymce/tinymce/blob/master/src/core/main/ts/EditorSettings.ts#L27

**Comment by lukesalter** [comment]
Yep found that array yesterday and recompiled my own version for now which appears to be working fine.
If anyone else is trying this: image, link and table all appear to work fine without a hitch on iOS when using the silver theme.

**Comment by TheSpyder** [comment]
This task has been scheduled, work is proceeding well and it should hit master in the next week or two.

**Comment by lnewson** [comment]
This issue has been fixed in TinyMCE 5.0.13, which has just been released to the community! Please let us know if you have any further issues.
